audio/alaw.c

```diff
@@ -0,0 +1,217 @@
+/**
+ * @file
+ * @author [sunzhenliang](https://github.com/HiSunzhenliang)
+ * @brief A-law algorithm for encoding and decoding (16bit pcm <=> a-law).
+ * This is the implementation of [G.711](https://en.wikipedia.org/wiki/G.711)
+ * in C.
+ **/
+
+/**
+ * Linear input code | Compressed code | Linear output code
+ * ------------------+-----------------+-------------------
+ * s0000000abcdx     | s000abcd        | s0000000abcd1
+ * s0000001abcdx     | s001abcd        | s0000001abcd1
+ * s000001abcdxx     | s010abcd        | s000001abcd10
+ * s00001abcdxxx     | s011abcd        | s00001abcd100
+ * s0001abcdxxxx     | s100abcd        | s0001abcd1000
+ * s001abcdxxxxx     | s101abcd        | s001abcd10000
+ * s01abcdxxxxxx     | s110abcd        | s01abcd100000
+ * s1abcdxxxxxxx     | s111abcd        | s1abcd1000000
+ *
+ * Compressed code: (s | eee | abcd)
+ **/
+#include <assert.h>    /// for assert
+#include <inttypes.h>  /// for appropriate size int types
+#include <stdio.h>     /// for IO operations
+
+/* length of test inputs */
+#define LEN ((size_t)8)
+
+/* input pcm for test */
+int16_t pcm[LEN] = {1000, -1000, 1234, 3200, -1314, 0, 32767, -32768};
+
+/* result coded alaw for test */
+uint8_t r_coded[LEN] = {250, 122, 230, 156, 97, 213, 170, 42};
+
+/* result decoded for test */
+int16_t r_decoded[LEN] = {1008, -1008, 1248, 3264, -1312, 8, 32256, -32256};
+
+/**
+ * @brief 16bit pcm to 8bit alaw
+ * @param out unsigned 8bit alaw array
+ * @param in  signed 16bit pcm array
+ * @param len length of pcm array
+ * @returns void
+ */
+void encode(uint8_t *out, int16_t *in, size_t len)
+{
+    uint8_t alaw;
+    int16_t pcm;
+    int32_t sign;
+    int32_t abcd;
+    int32_t eee;
+    int32_t mask;
```

Comment:
You may want to initialize these variables to zero.
```suggestion
    uint8_t alaw = 0;
    int16_t pcm = 0;
    int32_t sign = 0;
    int32_t abcd = 0;
    int32_t eee = 0;
    int32_t mask = 0;
```